lm_eval/models/huggingface.py

```diff
@@ -224,9 +225,8 @@ def __init__(
             else:
                 self.tokenizer.add_special_tokens({"pad_token": "<|pad|>"})
 
-        # TODO: override this for Gemma
         self.add_bos_token = add_bos_token
-        if getattr(self.config, "model_type", None) == "gemma":
+        if getattr(self.config, "model_type", None) in ["gemma", "recurrentgemma", "gemma2"]:
```

Comment:
should be `if "gemma" in getattr(self.config, "model_type", None):`

Reply:
Pushed a [wip] commit, to check for specific all these Gemma variants but in a more maintainable way. Or I can remove `recurrentgemma` and `gemma2` and make something like this:
```
self.add_bos_token = add_bos_token
model_type = getattr(self.config, "model_type", None)
if model_type and "gemma" in model_type.lower():
    self.add_bos_token = True
    eval_logger.info(
        f"Model type is '{self.config.model_type}', a BOS token will be used as Gemma underperforms without it."
    )
```

Reply:
this is fixed by the merge I think